modules/resources/resource-group/main.bicep

```
@@ -51,25 +51,32 @@ resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (ena
   }
 }
 
-resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = if (managedBy == null) {
   location: location
   name: name
   tags: tags
-  managedBy: managedBy
+  properties: {}
+}
+
+resource managedResourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = if (managedBy != null) {
+  location: location
+  name: name
+  tags: tags
+  managedBy: length(managedBy ?? '') > 0 ? managedBy : ''
```

```suggestion
  managedBy: managedBy ?? ''
```
If I'm not mistaken, this should do the trick to. Even better would be only `mangedBy: mangedBy`, IF `null` is accepted by the RP.

---

That is correct, but the RP does not accept `null`. If its json property exists it has to be a string, which makes this problem hard. This is why I've had challenges figuring out who to report this to CARML/AVM, Bicep, or Azure Support for RP.

I think the RP should accept null and require a string length greater than 0. Regardless it's a breaking change if someone used this module to create the resource group because it has to be deleted to remove managedBy AFAIK.

---

In the corresponding AVM module the owner now opted to not expose the ability to write to this property. This won't impact the RP but at least it wouldn't cause any other issues for users of the module.